src/interface/IVault.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.24;
+
+interface IVault {
+    error ZeroAddress();
+    error InvalidString();
+    error InvalidArray();
+    error ExceededMaxDeposit();
+    error InvalidAsset();
+    error InvalidDecimals();
+    error AssetNotFound();
+    error Paused();
+
+    event DepositAsset(address indexed asset, address indexed vault, uint256 amount, address indexed receiver);
+    event SetRateProvider(address indexed rateProvider);
+    event AddAsset(address indexed asset, uint256 decimals, uint256 index);
+    event ToggleAsset(address indexed asset, bool active);
+    event Pause(bool paused);
+    event Deposit(address indexed caller, address indexed owner, uint256 assets, uint256 shares);
+
+    struct VaultStorage {
+        uint256 totalAssets;
+        // QUESTION: Why was the enzyme 
+        address rateProvider;
+        bool paused;
+    }
+
+    struct AssetParams {
+        bool active;
+        uint256 index;
+        uint8 decimals;
+        uint256 idleAssets;
+        uint256 deployedAssets;
+    }
+
+    struct AssetStorage {
+        mapping(address => AssetParams) assets;
+        address[] list;
+    }
+
+    struct StrategyParams {
+        bool active;
+        uint256 index;
+        uint256 ratio;
```

Comment:
if we allow the governor/strategy controller to do arbitrary strategy allocations

do we still need the ratio ?

Reply:
ratio has been removed. see the processor function